### PR TITLE
Naive hungarian

### PR DESCRIPTION
**Pull request by harel55**
Create naive implementation of Hungarian algorithm for assigning alephs to drivers.

The algorithm works with a cost matrix of rows of agents and columns of tasks, and finds the bijection from agents to tasks that minimizes the sum of the costs, which in this case are the distances from each driver to each aleph. In order to deal with the fact that each agent will only be assigned one task, I made each available seat in a driver's car an "agent".

The algorithm is naive to the extent that it assumes each driver returns home before dropping off their next assigned aleph. This is suboptimal, but should do fairly well overall.

If there are more spots than alephs in need of rides, the algorithm will act normally. If vice versa, the alephs left unassigned will remain in the **list** alephsToOptimize

### CODE REVIEW

File: app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java [hunk 1]
```diff
@@ -13,21 +14,27 @@
 public class RidesOptimizer {
 
     /**
+     * Calculate rides based on latitude and longitude, finding the optimal assignment of passengers
+     * to drivers in order to minimize total distance traveled by all drivers. Makes the simplifying
+     * assumption that all drivers return to their home in between each drop-off.
+     */
+    public static final int ALGORITHM_NAIVE_HUNGARIAN=0;
```
Comment (ischeinkman) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java]: Why did you move all of the other algorithms down instead of adding yours to the end?

File: app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java [hunk 2]
```diff
@@ -13,21 +14,27 @@
...
-    public static final int ALGORITHM_LATLONG_DRIVERS_FIRST=1;
-    private Set<ContactInfoWrapper> alephsToOptimize;
+    public static final int ALGORITHM_LATLONG_DRIVERS_FIRST=2;
+    private List<ContactInfoWrapper> alephsToOptimize;
     private List<DriverInfoWrapper> driversToOptimize;
     private int algorithm;
     private boolean retainPreexisting;
     public RidesOptimizer(){
-        this.alephsToOptimize=new HashSet<ContactInfoWrapper>();
+        this.alephsToOptimize=new ArrayList<ContactInfoWrapper>();
```
Comment (ischeinkman) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java]: Is it really necessary to change this to an arraylist? Hashmap allows automatic removal of duplicates, which is an issue I ran into before.

File: app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java [hunk 3]
```diff
@@ -109,6 +119,354 @@ public void optimize(){
         }
     }
 
+    private void naiveHungarian() {
```
Comment (ischeinkman) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java:175]: New code should be added to the bottom, not the top.

File: app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java (hunk 2 above)
Comment (harel55) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java]: It was necessary for me to be able to associate indices with the elements. I did notice the issue with duplicates. If this is the cause, I can fix it when I have access to my desktop again.

File: app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java (hunk 1 above)
Comment (harel55) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java]: Because it was my intent to have this algorithm listed first. I didn't want to have the order on the pick list different from the organization of the code. If it bothers you, I'm fine with it being changed

Comment (ischeinkman) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java]: If possible I would prefer both picklist and code to be in order that the algorithms were added. Thank you.

File: app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java (hunk 2 above)
Comment (ischeinkman) [inline app/src/main/java/org/ramonaza/officialramonapp/people/rides/backend/RidesOptimizer.java]: Thank you.
